api/src/main/java/com/epam/pipeline/dao/monitoring/metricrequester/AbstractMetricRequester.java

```diff
@@ -151,6 +154,17 @@ public static MonitoringRequester getStatsRequester(final ELKUsageMetric metric,
         );
     }
 
+    public Map<String, Double> collectAggregation(final SearchResponse response,
+                                                  final String aggName, final String subAggName) {
+        return ((Terms)response.getAggregations().get(aggName)).getBuckets()
+                .stream()
+                .map(b -> Pair.of(b.getKey().toString(),
+                        doubleValue(b.getAggregations().asList(), subAggName))
```

Comment:
I would suggest to use `aggregations(b)` call here rather than `b.getAggregations().asList()`. It does just the same but is null-safe.